dot/peerset/peerstate.go

```diff
@@ -212,15 +215,16 @@ func (ps *PeersState) sortedPeers(idx int) peer.IDSlice {
 
 func (ps *PeersState) addReputation(pid peer.ID, change ReputationChange) (
 	newReputation Reputation, err error) {
-	ps.mu.Lock()
-	defer ps.mu.Unlock()
-
 	n, err := ps.getNode(pid)
```

Comment:
we don't need to lock before this call?

Reply:
The `PeersState` causes race condition into `network` pkg because the `getNode` function access the `ps.nodes` map at the same time `addReputation` updates the map. I added `Mutex` to `PeersState` struct so `getNode` uses the `Lock/Unlock` to lock while reading and after `getNode` executes `addReputation` locks to write. It is not possible locking after `getNode` since `getNode` will starve waiting the unlock that will never be called, makes sense? 😅 

However the previous implementation is not good so I updated to use a `RWMutex` instead and remove the `getNode` call from `addReputation` which uses the `Lock/Unlock` and now `getNode` uses `RLock/RUnlock`, basically `getNode` will blocks if another goroutine calls the `addReputation` first